aten/src/ATen/native/MaxPooling.cpp

```diff
@@ -98,6 +98,14 @@ Tensor max_pool1d(
     IntArrayRef padding,
     IntArrayRef dilation,
     bool ceil_mode) {
+
+  auto ndim = self.ndimension();
+  TORCH_CHECK(
+      (ndim == 2 && self.size(0) != 0 && self.size(1) != 0) ||
+          (ndim == 3 && self.size(1) != 0 && self.size(2) != 0),
+      "max_pool1d: Expected 2D or 3D (batch mode) tensor with optional 0 dim batch size for input, but got:",
+      self.sizes());
```

Comment:
We add the error here as `max_pool1d_with_indices` is implemented using `max_pool2d` and hence it returns incorrect error message : `'Expected 3D or 4D (batch mode) tensor with optional 0 dim batch size for input, but got:[17, 0, 1, 50]`